File: src/main/java/net/kigawa/kutil/unit/runtimeexception/NoSingleUnitException.java
package net.kigawa.kutil.unit.runtimeexception;

public class NoSingleUnitException extends RuntimeUnitException
{
    public NoSingleUnitException(String message, Throwable cause)
    {
        super(message, cause);
    }

    public NoSingleUnitException(String message)
    {
        super(message);
    }
}

File: kunit/src/jvmMain/kotlin/net/kigawa/kutil/kunit/api/exception/NoSingleUnitException.kt
package net.kigawa.kutil.kunit.api.exception

import net.kigawa.kutil.kunit.api.UnitIdentify
import net.kigawa.kutil.kunit.api.component.UnitInfo

class NoSingleUnitException(
  override val message: String?,
  override val cause: Throwable? = null,
  @Suppress("UNUSED_PARAMETER") identify: UnitIdentify<out Any>,
  @Suppress("UNUSED_PARAMETER") units: List<Any>? = null,
  @Suppress("UNUSED_PARAMETER") infoList: List<UnitInfo<out Any>>? = null,
): UnitException()